src/video_core/renderer_opengl/renderer_opengl.h

```diff
@@ -65,8 +65,8 @@ class RendererOpenGL : public VideoCore::RendererBase {
     void ConfigureFramebufferTexture(TextureInfo& texture,
                                      const Pica::FramebufferConfig& framebuffer);
     void DrawScreens(const Layout::FramebufferLayout& layout, bool flipped);
-    void ApplySecondLayerOpacity();
-    void ResetSecondLayerOpacity();
+    void ApplySecondLayerOpacity(bool isPortrait = false);
+    void ResetSecondLayerOpacity(bool isPortrait = false);
```

Comment:
Why does the second screen's opacity have anything to do with screen orientation?

Reply:
The `ApplySecondLayerOpacity` setting is only applied when a custom screen layout is being used. In the code right now, the way that is determined is simply looking at the layout option setting. But in the new code that leads to a potential issue where a user is using a custom layout in landscape mode (the traditional CustomLayout option or the original custom_layout setting) but not in portrait, and so opacity should not be applied in portrait mode.

For now, I have (in the implementation of that function) disabled the opacity setting entirely on portrait mode but the only way I could see for that function to _identify_ if it is portrait mode was to pass it as an argument. 

It is possible there is, or could be, some sort of cleaner more universal way for the c++ code to identify that the app is in portrait mode. Happy to discuss.

Reply:
Yes, but the problem I have is that I don't know if there is any good way for the openGL renderer to know which layout is being used, only which one is selected in the settings. But there will always be both a landscape and portrait layout selected in the settings, so the problem comes in when the custom landscape option is selected, with an opacity written, but we are in portrait mode so it should not be applied.

There is a static isPortrait method that belongs to emu_window.cpp - can the OpenGL renderer access that method? It isn't labeled as part of a class, so I couldn't figure out how to access it outside of that file. I think it's private to that  file (my c++ is a little fuzzy, I do much more java coding these days)

The calling function, drawScreens, has access to a FrameBufferLayout layout variable. Maybe we could add "isPortrait" as a field for the FrameBufferLayout class, and set when it the layout is created? I think its reasonable to do that now that we have differentiated the idea of portrait and landscape layouts, its certainly reasonalbe to consider it a property of the layout itself.

Reply:
I will resolve this conversation in favour of a review of the newly added code